lineax/_operator.py

```diff
@@ -1429,6 +1505,33 @@ def _(operator):
     return operator.diagonal, operator.lower_diagonal, operator.upper_diagonal
 
 
+@ft.singledispatch
+def woodbury(
+    operator: AbstractLinearOperator,
+) -> tuple[
+    AbstractLinearOperator,
+    Shaped[Array, " k k"],
+    Shaped[Array, " n k"],
+    Shaped[Array, " k n"],
+]:
+    """Extracts the A, C, U, V Woodbury structure, from a linear
+    operator. Returns one linear operators and three matrices.
+    **Arguments:**
+    - `operator`: a linear operator.
+    **Returns:**
+    A 4-tuple, consisting of
+    - A which is a linear operator
+    - C, U and V which are matrices
+    For all but the Woodbury operator this extraction is not possible
+    """
+    _default_not_implemented("woodbury", operator)
```

Comment:
I don't think we actually need this function!

For a `WoodburyLinearOperator` then a user can just access `.{A, C, U, V}` directly. For anything else then they can't extract these components at all.